java/src/main/java/ai/rapids/cudf/RegexProgram.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.rapids.cudf;
+
+/**
+ * This is the regex program class.
+ */
+public class RegexProgram {
+  private String pattern; /* regex pattern */
```

Comment:
In general, please consider switching to `//` for comments like these. (Doesn't apply to Javadocs, of course.)
Commenting code is easier without the `/* */` comments getting in the way.